src/main/java/org/primefaces/component/datatable/DataTableBase.java

```diff
@@ -229,11 +229,11 @@ public void setScrollRows(int scrollRows) {
         getStateHelper().put(PropertyKeys.scrollRows, scrollRows);
     }
 
-    public Object getRowKey() {
-        return getStateHelper().eval(PropertyKeys.rowKey, null);
+    public String getRowKey() {
```

Comment:
Does this mean RowKey can't be a Long anymore or a composite key?

Reply:
It can, it will be coerce into String. What do you mean by composite key?

Reply:
Composite key meaning if it was a Hibernate Entity it can have a PK class that is more than one field to make up its uniquness.
```java
public class MyEntity {

     @Id
     private MyPk myPrimaryKey;
```

Reply:
Then, it will work same as today your PK will be coerced into a String (and hopefully you overrided `PK#toString()`

Reply:
OK!